### PR TITLE
[script.keymap] 1.1.2+matrix.1

### PR DESCRIPTION
**Pull request by pkscout**
### Description

- Added support for v19 Matrix
- Fix so script only backups user defined keymaps if trying to save new keymap file
- Moved all text into language file for easier internationalization
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.keymap/resources/lib/editor.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            window = list(WINDOWS.keys())[idx]
+
+            while not KODIMONITOR.abortRequested():
+                # Select category menu
+                idx = Dialog().select(tr(30008), list(ACTIONS.keys()))
+                if idx == -1:
+                    break
+                category = list(ACTIONS.keys())[idx]
+
+                while True:
```
Comment (enen92) [inline script.keymap/resources/lib/editor.py:49]: forever running loop. Note that there isn't any pause on the loop which will for sure consume too much CPU while this method is in use

File: script.keymap/resources/lib/keymap.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            userkeymap = utils.read_keymap(gen_file)
+        except Exception:
+            traceback.print_exc()
+            utils.rpc('GUI.ShowNotification', title="Keymap Editor",
+                      message="Failed to load keymap file", image='error')
+            return
+
+    ## main loop ##
+    confirm_discard = False
+    while True:
```
Comment (enen92) [inline script.keymap/resources/lib/keymap.py:62]: forever running loop

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments regarding forever running loops

**Comment by pkscout** [comment]
> See comments regarding forever running loops

I'll look at it tomorrow. This same logic is in the Leia version, so you might want to undo that merge if you can. Otherwise I'll bump the version number on both branches and fix them at the same time. 

Interestingly, I didn't change that logic, so sometime way in the past it got approved. But I agree, it needs to be fixed.

File: script.keymap/resources/lib/editor.py (hunk 1 above)
Comment (pkscout) [reply script.keymap/resources/lib/editor.py]: I've added an abort request check to the loop.  I looked through the code again, and the select dialog boxes act as pauses in the loops, so there don't appear to be any race conditions here. I also tested by bringing up a dialog box and then quitting Kodi.  Kodi quit within the normal amount of time and the log didn't show any forced terminations of python threads.

File: script.keymap/resources/lib/keymap.py (hunk 2 above)
Comment (pkscout) [reply script.keymap/resources/lib/keymap.py]: see comment above

**Review by enen92 — APPROVED** [review]

**Pull request merged**